bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/internal/items/ItemRegistryImpl.java

```diff
@@ -342,63 +333,36 @@ protected void notifyListenersAboutUpdatedElement(Item oldElement, Item element)
-            super(context, StateDescriptionProvider.class.getName(), null);
-        }
-
-        @Override
-        public StateDescriptionProvider addingService(ServiceReference<StateDescriptionProvider> reference) {
-            StateDescriptionProvider provider = context.getService(reference);
-
-            Object serviceRanking = reference.getProperty(Constants.SERVICE_RANKING);
-            if (serviceRanking instanceof Integer) {
-                stateDescriptionProviderRanking.put(provider.getClass().getName(), (Integer) serviceRanking);
-            } else {
-                stateDescriptionProviderRanking.put(provider.getClass().getName(), 0);
-            }
-
-            synchronized (stateDescriptionProviders) {
-                stateDescriptionProviders.add(provider);
+    protected void addStateDescriptionProvider(StateDescriptionProvider provider) {
+        synchronized (stateDescriptionProviders) {
+            stateDescriptionProviders.add(provider);
 
-                Collections.sort(stateDescriptionProviders, new Comparator<StateDescriptionProvider>() {
-                    // sort providers by service ranking in a descending order
-                    @Override
-                    public int compare(StateDescriptionProvider provider1, StateDescriptionProvider provider2) {
-                        return stateDescriptionProviderRanking.get(provider2.getClass().getName())
-                                .compareTo(stateDescriptionProviderRanking.get(provider1.getClass().getName()));
-                    }
-                });
-
-                for (Item item : getItems()) {
-                    ((GenericItem) item).setStateDescriptionProviders(stateDescriptionProviders);
+            Collections.sort(stateDescriptionProviders, new Comparator<StateDescriptionProvider>() {
+                // sort providers by service ranking in a descending order
+                @Override
+                public int compare(StateDescriptionProvider provider1, StateDescriptionProvider provider2) {
+                    return provider2.getRank().compareTo(provider1.getRank());
                 }
-            }
-            return provider;
-        }
+            });
 
-        @Override
-        public void removedService(ServiceReference<StateDescriptionProvider> reference,
-                StateDescriptionProvider service) {
-            stateDescriptionProviders.remove(service);
-            stateDescriptionProviderRanking.remove(service.getClass().getName());
             for (Item item : getItems()) {
                 ((GenericItem) item).setStateDescriptionProviders(stateDescriptionProviders);
             }
         }
     }
 
+    protected void removeStateDescriptionProvider(StateDescriptionProvider provider) {
+        stateDescriptionProviders.remove(provider);
+        for (Item item : getItems()) {
+            ((GenericItem) item).setStateDescriptionProviders(stateDescriptionProviders);
```

Comment:
Not sure if it would make a difference for Items, as they would not react on that themselves anyway. 

However, it's the way it was before this change, I didn't touch that (apart from moving it from the tracker to the DS injection method...)